tests/models/test_lora_layers.py

```diff
@@ -212,3 +213,75 @@ def test_lora_save_load_legacy(self):
 
         # Outputs shouldn't match.
         self.assertFalse(torch.allclose(torch.from_numpy(orig_image_slice), torch.from_numpy(lora_image_slice)))
+
+    # copied from: https://colab.research.google.com/gist/sayakpaul/df2ef6e1ae6d8c10a49d859883b10860/scratchpad.ipynb
+    def get_dummy_tokens(self):
+        max_seq_length = 77
+
+        inputs = torch.randint(2, 56, size=(1, max_seq_length), generator=torch.manual_seed(0))
+
+        prepared_inputs = {}
+        prepared_inputs["input_ids"] = inputs
+        return prepared_inputs
+
+    def get_text_lora_attn_procs(self, text_encoder: nn.Module, randn_weight=False):
```

Comment:
A few things:

* Could we reuse this function (making changes to it is completely fine)?

https://github.com/huggingface/diffusers/blob/85eff637aad1106f593d7535ec41cdb736b0b2ea/tests/models/test_lora_layers.py#L46

* Also, from our discussions in https://github.com/huggingface/diffusers/pull/3437 (particularly this [comment](https://github.com/huggingface/diffusers/pull/3437#issuecomment-1549930848)), it seems we also need to change the target modules for which we're applying LoRA, no?

Reply:
> Could we reuse this function (making changes to it is completely fine)?

I missed the existence of this function. I have made changes to reuse some of it in this commit. [1da772b](https://github.com/huggingface/diffusers/pull/3490/commits/1da772b9fe8f64702989c1319348dfffd65dc491)

>Also, from our discussions in https://github.com/huggingface/diffusers/pull/3437 (particularly this https://github.com/huggingface/diffusers/pull/3437#issuecomment-1549930848), it seems we also need to change the target modules for which we're applying LoRA, no?

This modifying might result in losing compatibility with already serialized files and might also require changes to the training code, so it might be better to do it in a separate PR. I'm thinking about opening another draft PR for that.

Reply:
Ok! I just opened #3490